lib/models/product/models.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'models.freezed.dart';
+part 'models.g.dart';
+
+@freezed
+class Product with _$Product {
+  const factory Product({
+    int? id,
+    String? title,
+    double? price,
+    String? description,
+    String? category,
+    String? image,
+    Rating? rating,
+  }) = _Product;
```

Comment:
It would be better if we set @Default() value for fields wherever possible to prevent null checks